parser.y

```diff
@@ -420,19 +424,24 @@ decl: declaration_specifiers init_declarator_list_opt SEMICOLON {
       }
     } else {
       auto decl = std::move(std::get<std::unique_ptr<DeclNode>>(decl_specifiers));
-      // A record declaration that doesn't declare any identifier, e.g., `struct point {int x, int y};`.
-      if (init_decl_list.empty()) {
-        decl_list.push_back(std::move(decl));
-      } else {
-        auto& rec_decl = dynamic_cast<RecordDeclNode&>(*decl);
-        // Initialize record variable.
+      auto& rec_decl = dynamic_cast<RecordDeclNode&>(*decl);
+
+      if (!init_decl_list.empty()) {
         for (auto& init_decl : init_decl_list) {
           if (init_decl) {
             init_decl->type = ResolveType(rec_decl.type->Clone(), std::move(init_decl->type));
           }
           decl_list.push_back(std::move(init_decl));
         }
       }
+
+      auto rec_type_id = dynamic_cast<RecordType&>(*rec_decl.type).id();
+      // TODO: handle cases where struct and union variables have the same name.
+      // Insert RecordDeclNode if it is a newly create type.
+      if (auto existing_type = custom_types.find(rec_type_id); existing_type == custom_types.end()) {
```

Comment:
I believe we can remove the temporary variable to make it an idiomatic _contains_ operation.
```suggestion
      if (custom_types.find(rec_type_id) == custom_types.end()) {
```

Reply:
Fixed!